File: Concurrent_Toy_Language_Interpreter/Model/Statements/CompStatement.java
package Model.Statements;

import ADTs.IStack;
import Model.ProgramState;

public class CompStatement implements IStatement {
    private IStatement first;
    private IStatement second;

    public CompStatement(IStatement one, IStatement two) {
        this.first = one;
        this.second = two;
    }

    public String toString() {
        return "(" + first.toString() + "; " + second.toString() + ")";
    }

    public ProgramState execute(ProgramState state) {
        IStack<IStatement> stack = state.getExeStack();
        stack.push(this.second);
        stack.push(this.first);
        return null;
    }
}

File: Toy_Language_Interpreter_C/Statements/CompStatement.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace a7
{
    class CompStatement : IStatement
    {
        private IStatement first;
        private IStatement second;

        public CompStatement(IStatement one, IStatement two)
        {
            this.first = one;
            this.second = two;
        }

        public ProgramState Execute(ProgramState state)
        {
            state.ExeStack.Push(this.second);
            state.ExeStack.Push(this.first);
            return state;
        }

        public override string ToString()
        {
            return "(" + this.first.ToString() + ";" + this.second.ToString() + ")";
        }
    }
}
